lib/libkontain/libkontain_mgmt.h

```diff
@@ -0,0 +1,26 @@
+#ifndef __LIBKONTAIN_MGMT_H__
+#define __LIBKONTAIN_MGMT_H__
+
+#define KM_MGMT_REQ_SNAPSHOT	1234
+
+/*
+ * Send this structure in the unix socket to the km management thread
+ * when trying to create a payload snapshot.
+ */
+#define SNAPLABELMAX 256
+#define SNAPDESCMAX 256
+#define SNAPPATHMAX 1024
+struct snapshot_req {
+   int opcode;				// what mgmt request is this.
+   int length;				// length of this request including opcode and the length
```

Comment:
Not sure why this is needed - the struct size is fixed, so `sizeof()` will do.

Reply:
I assume there will be other mgmt operations we want to support in the future.
If/when that happens then I would become less slothful and turn this into what it should be:
struct mgmt_request {
   int opcode;
   int length;
   union {
      struct snapshot_request {
         blah blah ...
      } snapshot;
      struct halt_and_catch_file_request {
        oh no oh no ....
      } hcf_req;
      .....
   } request_types;
};